src/components/DatasetSelector/DatasetSelector.tsx

```diff
@@ -0,0 +1,187 @@
+import { Dataset } from "@/service/types"
+
+import { styled, lighten, darken } from '@mui/system';
+
+const GroupHeader = styled('div')(({ theme }) => ({
+  position: 'sticky',
+  top: '-8px',
+  padding: '4px 10px',
+  color: theme.palette.primary.main,
+  backgroundColor:
+    theme.palette.mode === 'light'
+      ? lighten(theme.palette.primary.light, 0.85)
+      : darken(theme.palette.primary.main, 0.8),
+}));
+
+const GroupItems = styled('ul')({
+  padding: 0,
+});
+
+
+function DatasetSelector({ datasetsList, setParentDataset, enableVersionSelector = true }) {
+
+  const [versions, setVersions] = useState(0)
+  const [filteredDatasetsList, setFilteredDatasetsList] = useState<Dataset[]>([])
+  const [layerFilteredDatasetsList, setLayerFilteredDatasetsList] = useState<Dataset[]>([])
+  const [layer, setLayer] = useState<string>('')
+  const [domain, setDomain] = useState<string>('')
+  const [dataset, setDataset] = useState<Dataset>(null)
+
+  useEffect(() => {
+    if (datasetsList) {
+      if (layer) {
+        if (domain) {
+          setFilteredDatasetsList(
+            datasetsList.filter((dataset) => dataset.layer == layer && dataset.domain == domain)
+          )
+        }
+        else {
+          setFilteredDatasetsList(
+            datasetsList.filter((dataset) => dataset.layer == layer)
+          )
+        }
+        setLayerFilteredDatasetsList(
+          datasetsList.filter((dataset) => dataset.layer == layer)
+        )
+      }
+      else {
+        setFilteredDatasetsList(datasetsList)
+        setLayerFilteredDatasetsList(datasetsList)
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [layer, domain])
+
+  useEffect(() => {
+    if (datasetsList) {
+      setFilteredDatasetsList(datasetsList)
+      setLayerFilteredDatasetsList(datasetsList)
+    }
+  }, [datasetsList])
+
+  const handleDomainSelect = (value) => {
+    if (value) {
+      const splits = value.split('/')
+      const layer = splits[0]
+      const domain = splits[1]
+      setLayer(layer)
+      setDomain(domain)
+    }
+    else {
+      setDomain(null)
+    }
+    setDataset(null)
+  }
+
+  const handleLayerSelect = (value) => {
+    if (value) { setLayer(value); setDomain(null) }
+    else { setLayer(null), setDataset(null), setDataset(null) }
+  }
+
+  useEffect(() => {
+    let version = 0
+    if (dataset) {
+      version = dataset.version
+      setLayer(dataset.layer)
+      setDomain(dataset.domain)
+    }
+    else {
+      setLayer(null)
+      setDomain(null)
+    }
+    setVersions(version)
+    setParentDataset(dataset)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [dataset])
+
+  return (
+    <>
+      <Row>
+        <div>
+          <Typography variant="h2">Layer</Typography>
+          <FormControl fullWidth sx={{ m: 1 }}>
+            <Autocomplete
+              multiple={false}
+              options={[...new Set(datasetsList.map((dataset) => dataset.layer))]}
```

Comment:
I think for the options logic we want to split this out into a function e.g. for this `getUniqueLayers()` and have this option just call this function